### ISSUE TITLE
Feature: picture support

### ISSUE THREAD

**Issue by patric-eberle** [alexjoverm/v-lazy-image#17]
Your plugin does a great job with `srcset` and `sizes`. Unfortunately we also need the `picture` element in our projects to display screen optimized images in different ratios. Do you have plans to add support for `picture` and `source`?

http://scottjehl.github.io/picturefill/#using-picture

**Comment by alexjoverm** [comment]
Hi! Sorry for the delay. Yeah that's something we could add indeed, although it will probably need some rework.

What could be a good way to do it while keep it simple for the user? Probably the best is to move the logic into a mixin, and create a different component for the picture based one?

Something like:

```
import {vLazyPicture} from 'v-lazy-image'
```

Let's give it a thought

**Comment by andreynazarov3** [comment]
I made this support in my project, though i didn't publish it yet as it's little too dirty right now.

**Comment by andreynazarov3** [comment]
I made this support in my project, though i didn't publish it yet as it's little too dirty right now. 
I use picture tag if srcset is provided:
```render(h) {
    if (this.srcset) {
      return h("picture", {
        class: {
          "v-lazy-image": true,
          "v-lazy-image-loaded": this.loaded
        }
      }, [
        h('source', {
          attrs: {
            media: "(max-width: 480px)",
            srcset: this.srcsetImage[2]
          }
        }),
        h('source', {
          attrs: {
            media: "(max-width: 1024px)",
            srcset: this.srcsetImage[1]
          }
        }),
        h('source', {
          attrs: {
            media: "(min-width: 1025px)",
            srcset: this.srcsetImage[0]
          }
        }),
        h('img', {
          attrs: {
            src: this.srcImage,
          },
          ref: 'image'
        })
      ]);
    } else {
      return h("img", {
        attrs: {
          src: this.srcImage,
          srcset: this.srcsetImage[0]
        },
        class: {
          "v-lazy-image": true,
          "v-lazy-image-loaded": this.loaded
        },
        ref: 'image'
      });
    }
  },```

**Comment by AlexeyShaykov** [comment]
Hi(привет!) @andreynazarov3. Thanks for your work. 
Whats news, about  picture support? i have same issues. 
Thanks.

**Comment by andreynazarov3** [comment]
@AlexeyShaykov , Hi, not yet, but it is will be soon, this weekend probably

**Comment by alexjoverm** [comment]
Thanks for the support on this feature! It'd be great if the feature keeps the API without any breaking changes. The `usePictureTag` props sounds good, any other ideas? The aim is to make it as flexible as posible, while keeping it concise and easy to use

**Comment by alexjoverm** [comment]
Even a better way, as suggested on #21, is to allow using scoped slots. In that way, we can provide a default way to use it (as it currently works) while providing the users the capability to render the component as they want, so that they could use a `picture` tag, a `canvas` or whatever while encapsulating the lazy-loading logic

**Comment by andreynazarov3** [comment]
@AlexeyShaykov  Well, I finally re-written component for `<picture>`, you can check out code here:
https://github.com/andreynazarov3/VueLazyPicture
I prefer using single-file component and included medium blur in it.
@alexjoverm I used your part for intersection observer. Everything else is new so it's not suitable for pull request.

**Comment by andreynazarov3** [comment]
Also I found out that vue has @load event you can place on image, very handy.

**Comment by AlexeyShaykov** [comment]
@andreynazarov3, Thx work awesomely!

**Comment by anthonygore** [comment]
Hey Alex. Another feature that would be useful would be to allow multiple source tags. That way you can use a modern image like webp and have a fallback to jpg e.g.

```
<picture>
  <source srcset="img/awesomeWebPImage.webp" type="image/webp">
  <source srcset="img/creakyOldJPEG.jpg" type="image/jpeg"> 
  <img src="img/creakyOldJPEG.jpg" alt="Alt Text!">
</picture>
```

**Comment by alexjoverm** [comment]
@anthonygore yes! This feature would allow that, thanks for the link as well.

Right now I don't have the time to work on this feature, but if anyone wants to I'm more than happy to review it!

**Comment by anthonygore** [comment]
Hey all, I think I've got a solution for this, would appreciate any feedback. 

Pull request #35

**Comment by alexjoverm** [comment]
Thanks to the amazing work of @anthonygore, this is available from 1.3.0 🎉
